pyvista/plotting/colors.py

```diff
@@ -344,17 +354,539 @@
+    'pv': 'paraview_background',
+    'paraview': 'paraview_background',
+    'slategrey': 'slategray',
+}
+
+
+class Color:
+    """Helper class to convert between different color representations used in the pyvista library.
+
+    Many pyvista methods accept :data:`color_like` parameters. This helper class
+    is used to convert such parameters to the necessary format, used by
+    underlying (VTK) methods. Any color name (``str``), hex string (``str``)
+    or RGB(A) sequence (``tuple``, ``list`` or ``numpy.ndarray`` of ``int``
+    or ``float``) is considered a :data:`color_like` parameter and can be converted
+    by this class.
+    See :attr:`Color.name` for a list of supported color names.
+
+    Parameters
+    ----------
+    color : color_like, optional
+        Either a string, RGB sequence, RGBA sequence, or hex color string.
+        RGB(A) sequences should either be provided as floats between 0 and 1
+        or as ints between 0 and 255. Hex color strings can contain optional
+        ``'#'`` or ``'0x'`` prefixes. If no opacity is provided, the
+        ``default_opacity`` will be used. If ``color`` is ``None``, the
+        ``default_color`` is used instead.
+        The following examples all denote the color 'white':
+
+        * ``'white'``
+        * ``'w'``
+        * ``[1.0, 1.0, 1.0]``
+        * ``[255, 255, 255, 255]``
+        * ``'#FFFFFF'``
+
+    opacity : int, float or str, optional
+        Opacity of the represented color. Overrides any opacity associated
+        with the provided ``color``. Allowed opacities are floats between 0
+        and 1, ints between 0 and 255 or hexadecimal strings of length 2
+        (plus the length of the optional prefix).
+        The following examples all denote a fully opaque color:
+
+        * ``1.0``
+        * ``255``
+        * ``'#ff'``
+
+    default_color : color_like, optional
+        Default color to use when ``color`` is ``None``. If this value is
+        ``None``, then defaults to the global theme color. Format is
+        identical to ``color``.
+
+    default_opacity : int, float or str, optional
+        Default opacity of the represented color. Used when ``color``
+        does not specify an opacity and ``opacity`` is ``None``. Format
+        is identical to ``opacity``.
+
+    Notes
+    -----
+    The internally used representation is an integer RGBA sequence (values
+    between 0 and 255). This might however change in future releases.
+
+    Examples
+    --------
+    Create a transparent green color using a color name, float RGBA sequence,
+    integer RGBA sequence and RGBA hexadecimal string.
+
+    >>> import pyvista
+    >>> pyvista.Color("green", opacity=0.5)
+    Color(name='green', hex='#00800080')
+    >>> pyvista.Color([0.0, 0.5, 0.0, 0.5])
+    Color(name='green', hex='#00800080')
+    >>> pyvista.Color([0, 128, 0, 128])
+    Color(name='green', hex='#00800080')
+    >>> pyvista.Color("#00800080")
+    Color(name='green', hex='#00800080')
```

Comment:
I don't have any suggested changes. However, since this is an Example of colors, I still think it would be more user-friendly if the color table could be plotted here. Of course, at this point, I think it still achieves its purpose as an Example.

Reply:
I thought about that, but as it is a very long table I wasn't sure if it would be an improvement, as it would push the Color class' methods and attributes all the way down in the docs. I will play around with this and see what can be done (maybe a collapsible table would be a good idea, if possible?)

Reply:
Yeah, it might be better to sneak in an explicit link to `Color.name` where the colour table is.

Reply:
I put an expandable table under the Notes section, let me know if that is ok.

Reply:
Great. Thanks.